SeniorSlideshowProject/src/AddAudioListener.java

```diff
@@ -0,0 +1,42 @@
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.Vector;
+
+import javax.swing.ImageIcon;
+import javax.swing.JFileChooser;
+
+import Slides.SlideShowStateMachine;
+import Slides.SlideState;
+
+public class AddAudioListener implements ActionListener {
+
+	private static final int FILES_AND_DIRECTORIES = 2;
+	@Override
+	public void actionPerformed(ActionEvent e) {
+		// TODO Auto-generated method stub
+		JFileChooser fc = new JFileChooser();
+		fc.setFileFilter(new FileTypeFilter(".jpg", "audio file"));
+		fc.setFileSelectionMode(FILES_AND_DIRECTORIES);
+		SlideShowStateMachine mySlideShow = SlideShowStateMachine.getInstance();
+		int result = fc.showDialog(null, "Add Audio");
+		if (result == JFileChooser.APPROVE_OPTION){
+			String selectedFilePath = fc.getSelectedFile().getPath();
+			if (selectedFilePath.contains(".jpg")){
+				ImageIcon icon = new ImageIcon(selectedFilePath);
+				SlideState mySlide = new SlideState(icon);
```

Comment:
Why would you have an addAudioListener that is adding slides to the state machine instead of audio files? This doesn't make sense to me.